hcc004v1_dysrhythmia_suspect.py

```diff
@@ -0,0 +1,318 @@
+        "444249",
+        "444944",
+        "444944",
+        "449494",
+        "449496",
+        "451558",
+        "451559",
+        "451560",
+        "453457",
+        "453462",
+        "454178",
+        "454180",
+        "454181",
+        "454205",
+        "454206",
+        "454207",
+        "454371",
+        "545231",
+        "545231",
+        "545232",
+        "545233",
+        "545238",
+        "545239",
+        "545239",
+        "558741",
+        "558745",
+        "559416",
+        "560050",
+        "563304",
+        "563305",
+        "563306",
+        "563310",
+        "564459",
+        "564460",
+        "565068",
+        "565069",
+        "573523",
+        "583982",
+        "583982",
+        "583985",
+        "583985",
+        "590326",
+        "590375",
+        "590376",
+        "591479",
+        "592349",
+        "592421",
+        "594710",
+        "594714",
+    }
+
+
+class Protocol(BaseProtocol):
+    """Dysrhythmia Suspects."""
+
+    # TODO: Add ProtocolOverride event types
+    RESPONDS_TO = [
+        EventType.Name(EventType.CONDITION_CREATED),
+        EventType.Name(EventType.CONDITION_UPDATED),
+        EventType.Name(EventType.MEDICATION_LIST_ITEM_CREATED),
+        EventType.Name(EventType.MEDICATION_LIST_ITEM_UPDATED),
+    ]
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.patient_id = self.patient_id_from_event()
+
+    # TODO: Getting the patient or patient ID from an objects seems like something that will happen
+    #  a lot. Should this be standardized in a base class? On the flip side, the fields needed from
+    #  a patient may not be standard. Maybe a simple, standard, efficient way to get a patient ID
+    #  from a target is a good place to start.
+    def patient_id_from_event(self) -> str:
+        """Get the patient ID from the event target."""
+        # TODO: Add cases for ProtocolOverride
+        match self.event.type:
+            case EventType.CONDITION_CREATED | EventType.CONDITION_UPDATED:
+                model = Condition
+            case EventType.MEDICATION_LIST_ITEM_CREATED | EventType.MEDICATION_LIST_ITEM_UPDATED:
+                model = Medication
+            case _:
+                raise AssertionError(
+                    f"Event type {self.event.type} not supported by 'patient_id_from_event'"
+                )
+
+        return (
+            model.objects.select_related("patient")
+            .values_list("patient__id")
+            .get(id=self.event.target)[0]
+        )
+
+    def in_denominator(self) -> bool:
+        """
+        Patients with any active medication in Antiarrhythmics Drug Class.
+        """
+        # TODO: Settable time frame is not currently supported — where would it come from?
+        time_frame_provided = False
+        if not time_frame_provided:
+            # TODO: Should these datetimes be based on local time or UTC time?
```

Comment:
@mbiannaccone

Reply:
i believe everything stored in the database is utc time and then on the front-end we convert to the local time, so it would probably make sense to do all the time calculations like this in utc time as well.